Seems like `SDL2main` cause `gtest` can't work

### Problem

gtest can't compile after #161 (I guess la)
It can compile after I removed `SDL2::SDL2main`
https://github.com/ntut-open-source-club/practical-tools-for-simple-design/blob/0399023f675cb7a59ed0f26ea7bacdea66ef0fae/CMakeLists.txt#L116-L119

### Steps to reproduce

1. `cmake --build build --target Tests`
2. got this
```
scoop/apps/gcc/current/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: _deps/sdl2-build/libSDL2main.a(SDL_windows_main.c.obj):SDL_windows_ma:(.text+0x15d): undefined reference to `SDL_main'
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### Expected behavior

It can compile

### PTSD version

commit 0399023f675cb7a59ed0f26ea7bacdea66ef0fae

### Operating system/version

Windows 11

### CMake version

Sorry IDK

### CMakeCache.txt

[CMakeCache.txt](https://github.com/ntut-open-source-club/practical-tools-for-simple-design/files/14946840/CMakeCache.txt)


### OpenGL Info

```txt
It can't even run :(
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,7 +114,6 @@ add_library(PTSD STATIC
     ${INCLUDE_FILES}
 )
 target_link_libraries(PTSD
-    SDL2::SDL2main # IDK what happens if we don't link SDL2main
     ${DEPENDENCY_LINK_LIBRARIES}
 )
 target_include_directories(PTSD SYSTEM PRIVATE
@@ -132,11 +131,14 @@ endif()
 target_compile_options(PTSD PRIVATE
     ${TARGET_COMPILE_OPTIONS}
 )
+set(DEPENDENCY_INCLUDE_DIRS ${DEPENDENCY_INCLUDE_DIRS} PARENT_SCOPE)
+set(DEPENDENCY_LINK_LIBRARIES ${DEPENDENCY_LINK_LIBRARIES} PARENT_SCOPE)
 
 add_executable(Example EXCLUDE_FROM_ALL
     ${EXAMPLE_FILES}
 )
 target_link_libraries(Example
+    SDL2::SDL2main # IDK what happens if we don't link SDL2main
     PTSD
 )
 target_include_directories(Example SYSTEM PRIVATE

